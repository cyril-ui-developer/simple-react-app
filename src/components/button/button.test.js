
import React from 'react';
import { shallow } from 'enzyme';
import Button from './button';


describe('Button', () => {
  it('should be defined', () => {
    expect(Button).toBeDefined();
  });
  it('should render name props correctly', () => {
    const component = shallow(
      <Button name="button one" />,
    );
    expect(component).toMatchSnapshot();
  });
  it('should render value props correctly', () => {
    const component = shallow(
      <Button value="buttonone" />,
    );
    expect(component).toMatchSnapshot();
  });
});
it('should have a button required props - name and value', () => {
  const component = shallow(<Button name="button one" value="buttonone" />);
  expect(component.props().name).toBe('button one');
  expect(component.props().value).toBe('buttonone');
});

it('should call mock function when button is clicked', () => {
  const mockFn = jest.fn();
  const component = shallow(
    <Button name="button test" handleClick={mockFn} />,
  );
  component.simulate('click');
  expect(mockFn).toHaveBeenCalled();
});
