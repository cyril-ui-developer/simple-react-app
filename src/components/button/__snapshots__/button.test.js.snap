// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render name props correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    disabled={false}
    handleClick={[Function]}
    name="button one"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "",
      "disabled": false,
      "name": "button one",
      "onClick": [Function],
      "type": "submit",
      "value": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "",
        "disabled": false,
        "name": "button one",
        "onClick": [Function],
        "type": "submit",
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Button should render value props correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    disabled={false}
    handleClick={[Function]}
    value="buttonone"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "",
      "disabled": false,
      "name": undefined,
      "onClick": [Function],
      "type": "submit",
      "value": "buttonone",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "",
        "disabled": false,
        "name": undefined,
        "onClick": [Function],
        "type": "submit",
        "value": "buttonone",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
